azurerm/internal/services/storage/resource_arm_storage_account.go

```diff
@@ -245,6 +245,19 @@ func resourceArmStorageAccount() *schema.Resource {
 								},
 							},
 						},
+						"change_feed": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
+										Type:     schema.TypeBool,
+										Required: true,
+									},
+								},
+							},
+						},
```

Comment:
Given this is only an true/false value and unless it will change prior to moving out of preview should we make this a top level property `change_feed_enasbled`

Reply:
Hi @katbyte , Thanks for reviewing. 
Because the field `change_feed` is a preview field and have three status: `true`, `false`, `nil`. So we decided to use nest block (same as go-sdk).

Reference: 
1. [Document](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-change-feed?tabs=azure-portal)
2. [Reference we are following up](https://github.com/terraform-providers/terraform-provider-azurerm/blob/8dc7ecafca4cc76173fc43a11f24c66de08c869f/azurerm/internal/services/compute/virtual_machine_scale_set.go#L25)

Reply:
@lrxtom2 - i'm not sure what we get by nested it in a block like this? could we please move it the top level as `change_feed_enasbled`. thanks

Reply:
It's a great idea but it's a preview feature. If we default any value and user don't register the feature into his subscription will meet an error message below. So it may effect many end users.
`autorest/azure: Service returned an error. Status=<nil> Code="FeatureNotSupportedForAccount" Message="Change Feed is not supported for the account."`
So we refer field `ultra_ssd_enabled` in `virtual machine` to use nest block.

Reply:
@lrxtom2 the API should only be returning that error when trying to Enable that functionality - otherwise that's a bug in the API? From Terraform's perspective I'd agree this wants to be added at the top level as a flag